feature/gnoi/system/tests/supervisor_switchover_test/supervisor_switchover_test.go

```diff
@@ -162,29 +163,49 @@ func TestSupervisorSwitchover(t *testing.T) {
 	}
 }
 
-func findMatchedComponents(t *testing.T, dut *ondatra.DUTDevice, r *regexp.Regexp) []string {
+func findComponentsByType(t *testing.T, dut *ondatra.DUTDevice, cType telemetry.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT) []string {
 	components := dut.Telemetry().ComponentAny().Name().Get(t)
 	var s []string
 	for _, c := range components {
-		if len(r.FindString(c)) > 0 {
-			s = append(s, c)
+		lookupType := dut.Telemetry().Component(c).Type().Lookup(t)
+		if !lookupType.IsPresent() {
+			t.Logf("Component %s type is not found", c)
+		} else {
+			componentType := lookupType.Val(t)
+			t.Logf("Component %s has type: %v", c, componentType)
+
+			switch v := componentType.(type) {
+			case telemetry.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT:
+				if v == cType {
+					s = append(s, c)
+				}
+			default:
+				t.Fatalf("Expected component type to be a hardware component, got (%T, %v)", componentType, componentType)
+			}
 		}
 	}
 	return s
 }
 
-func findActiveRP(t *testing.T, dut *ondatra.DUTDevice, supervisors []string) (string, string) {
+func findStandbyRP(t *testing.T, dut *ondatra.DUTDevice, supervisors []string) (string, string) {
```

Comment:
Since this is named `findStandbyRP`, should this return just the standbyRP, or should it have a different name to be more accurate?

Reply:
Agreed. Since the similar function is also used in other test code, I plan to move them to a library as helper function and update the function name as well.